modules/dnf/dnf.tsp

```diff
@@ -0,0 +1,77 @@
+import "@typespec/json-schema";
+using TypeSpec.JsonSchema;
+
+@jsonSchema("/modules/dnf.json")
+model DnfModule {
+    /** The dnf module offers pseudo-declarative package and repository management using dnf.
+     * https://blue-build.org/reference/modules/dnf/
+     */
+    type: "dnf";
+
+    /** List of links to .repo files to download into /etc/yum.repos.d/. */
+    repos?: Array<string>;
+
+    /** List of links to key files to import for installing from custom repositories. */
+    keys?: Array<string>;
+
+    /** List of folder names under /opt/ to enable for installing into. */
+    optfix?: Array<string>;
+
+    /** Configuration of RPM groups removal. */
+    "group-remove"?: {
+        /** List of RPM groups to remove. */
+        packages?: string;
```

Comment:
```suggestion
        packages: Array<string>;
```
Should be an array and if `group-remove` is defined, we MUST have the list of packages